util/codec/decimal.go

```diff
@@ -31,6 +31,7 @@ func EncodeDecimal(b []byte, d types.Datum) []byte {
 	bin, err := dec.ToBin(precision, frac)
 	if err != nil {
 		log.Errorf("should not happen, precision %d, frac %d %v", precision, frac, err)
+		panic("expected error happened during codec")
```

Comment:
And we should let it panic when we meet an unexpected bug and don't want to return an error.